cdap-log-publisher-spi/src/main/java/io/cdap/cdap/logs/LogPublisher.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logs;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import java.io.Closeable;
+
+/**
+ * Publishes log events to a destination.
+ */
+public interface LogPublisher extends Closeable {
```

Comment:
I would recommend to override close method to throw no checked exceptions. This would make interface consistent. You can also inherit from AutoCloseable